sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java

```diff
@@ -1085,4 +1086,30 @@ private boolean validateNoETag(BlobRequestConditions modifiedRequestConditions)
         }
         return modifiedRequestConditions.getIfMatch() == null && modifiedRequestConditions.getIfNoneMatch() == null;
     }
+
+    private boolean validateNoTime(BlobRequestConditions modifiedRequestConditions) {
+        if (modifiedRequestConditions == null) {
+            return true;
+        }
+        return modifiedRequestConditions.getIfModifiedSince() == null
+            && modifiedRequestConditions.getIfUnmodifiedSince() == null;
+    }
+
+    Mono<Response<BlobContainerAsyncClient>> renameWithResponse(ContainerRenameOptions options, Context context) {
+        StorageImplUtils.assertNotNull("options", options);
+        BlobRequestConditions requestConditions = options.getRequestConditions() == null ? new BlobRequestConditions()
+            : options.getRequestConditions();
+        context = context == null ? Context.NONE : context;
+
+        if (!validateNoETag(requestConditions) || !validateNoTime(requestConditions)
+            || requestConditions.getTagsConditions() != null) {
+            throw logger.logExceptionAsError(new UnsupportedOperationException(
+                "Lease-Id is the only HTTP access condition supported for this API"));
```

Comment:
If Lease-Id is the only valid conditional for this operation why do we accept the more complex match conditions object? Is this to allow for extensibility in the future?

Reply:
Yeah I think it is extensibility. I have seen areas in the SDK where we only take in lease id as well though so there is no set in stone pattern we've stuck with

Reply:
@alzimmermsft Do you have a preference on how this gets exposed?

Reply:
We can always expose a String leaseId and deprecate the setter getter if more request conditions become valid

Reply:
I'd lean towards using the same deciding factor as when we opt for options parameter types, if this is expected to grow in the near term future I'd lean towards keeping code as-is.